stdlib/2/typing.pyi

```diff
@@ -352,64 +351,77 @@ class TextIO(IO[unicode]):
 class ByteString(Sequence[int]): ...
 
 class Match(Generic[AnyStr]):
-    pos = 0
-    endpos = 0
-    lastindex = 0
-    lastgroup = ...  # type: AnyStr
-    string = ...  # type: AnyStr
+    pos: int
+    endpos: int
+    lastindex: Optional[int]
+    string: AnyStr
 
     # The regular expression object whose match() or search() method produced
-    # this match instance.
-    re = ...  # type: 'Pattern[AnyStr]'
+    # this match instance. This should not be Pattern[AnyStr] because the type
+    # of the pattern is independent of the type of the matched string in
+    # Python 2. Strictly speaking Match should be generic over AnyStr twice:
+    # once for the type of the pattern and once for the type of the matched
+    # string.
+    re: Pattern[Any]
```

Comment:
I wonder what problems we'd end up getting when we actually made it generic in two variables? Since Match objects are only created by calling pat.match() we should have all the types available when the latter is called, right? I guess the real problem is the vast amount of code that's already declaring match variables as Match[something] -- those would all break. I guess we'll have to wait for a way to declare defaults for type variables...

Reply:
Yes, I think the main problem is that we'd have to change `typing.py` too.

It also doesn't seem that useful; the second type argument only matters if you access the `.re` attribute and care about whether that was a str or unicode pattern. I don't think that's very common.